src/test/java/de/qabel/desktop/crashReports/HockeyAppTest.java

```diff
@@ -0,0 +1,174 @@
+    private String email;
+    private String stacktrace;
+
+    public HockeyAppTest() {
+        versionClient.setVersion(new HockeyAppVersion(1, config.getAppVersion()));
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        feedback = "HockeyAppTest feedback text";
+        name = "HockeyAppTest";
+        email = "HockeyAppTest@desktop.de";
+        stacktrace = "XCEPTION REASON STRING\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "ANOTHER EXCEPTION REASON STRING\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)\n" +
+            "  at CLASS.METHOD(FILE:LINE)";
+
+        stubFeedbackResponse();
+        stubVersionToApp();
+    }
+
+
+    @Test
+    public void testSendFeedback() throws IOException {
+
+        hockeyApp.sendFeedback(feedback, name, email);
+        String responseContent = getStubFeedbackResponseContent();
+        JSONObject json = parseFeedbackJson(responseContent);
+
+        assertEquals(123, json.get("id"));
+        assertEquals(shortVersion, versionClient.getVersion().getShortVersion());
+    }
+
+
+    @Test
+    public void testSendCrashReport() throws IOException {
+        stubSendCrashReport();
+        hockeyApp.sendStacktrace(feedback, stacktrace);
+    }
+
+    @Test
+    public void createLog() throws IOException {
+        String containingString = "Version: 1.1";
+        stacktrace = hockeyApp.createLog(stacktrace);
+
+        assertTrue(stacktrace.contains(containingString));
+    }
+
+    private void stubSendCrashReport() {
+        String responseContent = "some foobar testing argh piares";
+
+        String uri = "https://rink.hockeyapp.net/api/2/apps/3b119dc227334d2d924e4e134c72aadc/crashes/upload";
+        CloseableHttpResponseStub response = this.createResponseFromString(200, responseContent);
+        httpClient.addResponse("POST", uri, response);
+    }
+
+    JSONObject parseFeedbackJson(String responseContent) throws IOException {
+        try {
+            JSONObject parsedJson = new JSONObject(responseContent);
+            return parsedJson.getJSONObject("feedback");
+        } catch (JSONException e) {
+            throw new IOException("returned JSON was invalid", e);
+        }
+    }
+
+    @Test
+    public void testBuildParams() throws IOException {
+
+        List<NameValuePair> parameters = hockeyApp.buildFeedbackParams(feedback, name, email);
+
+        String paramVersionKey = parameters.get(4).getName();
+        String paramVersion = parameters.get(4).getValue();
```

Comment:
This test is brittle. If somebody removes one parameter or adds one (which will be done because that "email" field seems to be doubled), one has to adjust these numbers. And of every of the parameters would be tested this way, one would have to adjust more and more parameters. So please write a method where you can get a parameter value by key. Then you can just do a 
```JAVA
assertEquals(1, params.get("app_version_id"))
```